### PR TITLE
Migrate 4.10 Testing for Weak Cryptography

### LINKED ISSUE #86: 4.10 Testing for weak Cryptography - v4 Migration Cleanup
Should be covered by: https://github.com/OWASP/OWASP-Testing-Guide-v5/pull/57

Someone should just do a sanity check.
- [ ] Sanity check

### PR DESCRIPTION
**Pull request by ThunderSon**
This PR closes #86 .

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

Each commit tackles a file, if that makes it easier to handle. The last commit updates the document 4.10.1 again, since I forgot to update from `-` to `*` for unordered lists.

### CODE REVIEW

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 1]
```diff
@@ -76,11 +78,11 @@ Also there are some attacks that can be used to intercept traffic if the web ser
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: Can we add direct links for references? Line number 67, 71 and 73.

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 2]
```diff
@@ -76,11 +78,11 @@ Also there are some attacks that can be used to intercept traffic if the web ser
 
 Various types of information which must be protected can be also transmitted in clear text. It is possible to check if this information is transmitted over HTTP instead of HTTPS. Please refer to specific tests for full details, for credentials [[3]] and other kind of data [[2]].
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: Can we add direct links for references?

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 3]
```diff
@@ -280,7 +282,7 @@ Nmap done: 1 IP address (1 host up) scanned in 8.64 seconds
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: `Example 3 Checking for Client-Initiated Renegotiation and Secure Renegotiation via Openssl (Manually)` - Title Case

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 4]
```diff
@@ -280,7 +282,7 @@ Nmap done: 1 IP address (1 host up) scanned in 8.64 seconds
 
 Openssl [[30]] can be used for testing manually SSL/TLS. In this example the tester tries to initiate a renegotiation by client `m` connecting to server with openssl. The tester then writes the fist line of an HTTP request and types “R” in a new line. He then waits for renegotiaion and completion of the HTTP request and checks if secure renegotiaion is supported by looking at the server output. Using manual requests it is also possible to see if Compression is enabled for TLS and to check for CRIME [[13]], for ciphers and for other vulnerabilities.
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: “R” in `R`

Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: Can we add direct links for references?

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 5]
```diff
@@ -572,74 +574,74 @@ Using "OpenSSL 1.0.2-beta1 24 Feb 2014" on
...
+no PFS available
 
 Done now (2014-04-17 15:07) ---> owasp.org:443 <---
 
-user@myhost: %    
+user@myhost: %
 ```
 
-STARTTLS would be tested via `testssl.sh -t smtp.gmail.com:587 smtp`, each ciphers with `testssl -e `<target>, each ciphers per protocol with `testssl -E `<target>. To just display what local ciphers that are installed for openssl see `testssl -V`. For a thorough check it is best to dump the supplied OpenSSL binaries in the path or the one of testssl.sh.
+STARTTLS would be tested via `testssl.sh -t smtp.gmail.com:587 smtp`, each ciphers with `testssl -e <target>`, each ciphers per protocol with `testssl -E <target>`. To just display what local ciphers that are installed for openssl see `testssl -V`. For a thorough check it is best to dump the supplied OpenSSL binaries in the path or the one of testssl.sh.
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: testssl.sh in backticks `testssl.sh`

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 6]
```diff
@@ -572,74 +574,74 @@ Using "OpenSSL 1.0.2-beta1 24 Feb 2014" on
...
 
 The interesting thing is if a tester looks at the sources they learn how features are tested, see e.g. Example 4. What is even better is that it does the whole handshake for heartbleed in pure /bin/bash with /dev/tcp sockets -- no piggyback perl/python/you name it.
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: File paths in backticks. `/bin/bash` , `/dev/tcp`

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 7]
```diff
@@ -1225,15 +1225,15 @@ Check the configuration of the web servers that provide https services. If the w
...
 Check the configuration on a Microsoft Windows Server (2000, 2003 and 2008) using the registry key:
 
-    HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\
+`HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\`
 
 that has some sub-keys including Ciphers, Protocols and KeyExchangeAlgorithms.
 
 #### Example 2. Apache
 
 To check the cipher suites and protocols supported by the Apache2 web server, open the ssl.conf file and search for the SSLCipherSuite, SSLProtocol, SSLHonorCipherOrder,SSLInsecureRenegotiation and SSLCompression directives.
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: `SSLCipherSuite`, `SSLProtocol`, `SSLHonorCipherOrder`, `SSLInsecureRenegotiation` and `SSLCompression`

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 8]
```diff
@@ -1205,13 +1205,13 @@ Inside corporate environments testers can see services that are not directly acc
...
 
-    $ socat TCP-LISTEN:9999,reuseaddr,fork PROXY:10.13.37.100:destined.application.lan:443,proxyport=3128
+`$ socat TCP-LISTEN:9999,reuseaddr,fork PROXY:10.13.37.100:destined.application.lan:443,proxyport=3128`
 
 Then the tester can target all other tools to localhost:9999:
 
-    $ openssl s_client -connect localhost:9999
+`$ openssl s_client -connect localhost:9999`
 
 All connections to localhost:9999 will be effectively relayed by socat via proxy to destined.application.lan:443.
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: `localhost:9999`

`destined.application.lan:443`

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 9]
```diff
@@ -1190,8 +1190,8 @@ The following is a scenario of how the attack can take place:
 
 To test if a website is vulnerable carry out the following tests:
 
-1.  Check if website supports both HTTP and HTTPS protocols
-2.  Check if cookies do not have the “Secure” flag
+1. Check if website supports both HTTP and HTTPS protocols
+2. Check if cookies do not have the “Secure” flag
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: `Secure` in backticks.

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 10]
```diff
@@ -1126,15 +1128,15 @@ Checking localhost:443 for TLS Compression support against CRIME (CVE-2012-4929)
 
 This tool [[39]] is most comprehensive SSL tests. It supports the following checks:
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: Direct links other than references.

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md [hunk 11]
```diff
@@ -1,24 +1,26 @@
+# Testing for Sensitive Information Sent Via Unencrypted Channels
+
 ## Summary
 
 Sensitive data must be protected when it is transmitted through the network. If data is transmitted over HTTPS or encrypted in another way the protection mechanism must not have limitations or vulnerabilities, as explained in the broader article "Testing for Weak SSL/TLS Ciphers, Insufficient Transport Layer Protection (OTG-CRYPST-001)" [[1]] and in other OWASP documentation [[2]], [[3]], [[4]], [[5]].
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md]: Can we add direct links for references?

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md [hunk 12]
```diff
@@ -123,11 +125,9 @@ Checking if logs or source code may contain phone number, email address, ID or a
 
 ## Tools
 
-- curl
-- grep
-- [Identity Finder](http://download.cnet.com/Identity-Finder-Free-Edition/3000-2144_4-10906766.html)
-
-
+* [curl](http://curl.haxx.se/)
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md]: https://curl.haxx.se/

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md [hunk 13]
```diff
@@ -123,11 +125,9 @@ Checking if logs or source code may contain phone number, email address, ID or a
...
+* [grep](http://man7.org/linux/man-pages/man1/egrep.1.html)
+* [Identity Finder](http://download.cnet.com/Identity-Finder-Free-Edition/3000-2144_4-10906766.html)
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md]: https://download.cnet.com/Identity-Finder-Free-Edition/3000-2144_4-10906766.html

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md [hunk 14]
```diff
@@ -123,11 +125,9 @@ Checking if logs or source code may contain phone number, email address, ID or a
...
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md]: [5]:http://curl.haxx.se/
duplicate below line 136

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md [hunk 15]
```diff
@@ -10,13 +12,13 @@ The testing guide is trying to provide a guideline how to identify the weak encr
...
-- When uses of RSA in encryption, Optimal Asymmetric Encryption Padding (OAEP) mode is recommended.
-- When uses of RSA in signature, PSS padding is recommended.
-- Weak hash/encryption algorithms should not be used such MD5, RC4, DES, Blowfish, SHA1. 1024-bit RSA or DSA, 160-bit ECDSA (elliptic curves), 80/112-bit 2TDEA (two key triple DES)
-- Minimum Key length requirement:
+* When the uses of AES128 and AES256, The IV (Initialization Vector) must be random and unpredictable. Refer to [FIPS 140-2, Security Requirements for Cryptographic Modules](http://csrc.nist.gov/cryptval/140-2.htm), section 4.9.1. random number generator tests. For example, in Java, `java.util.Random` is considered a weak random number generator. `java.security.SecureRandom` should be used instead of `java.util.Random`.
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md]: I think we have used `-` for lists in previous files. Can we make it unique everywhere?

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md [hunk 16]
```diff
@@ -119,31 +118,26 @@ CWE-329: Not Using a Random IV with CBC Mode
...
+* [Insecure Randomness](https://www.owasp.org/index.php/Insecure_Randomness)
+* [Insufficient Entropy](https://www.owasp.org/index.php/Insufficient_Entropy)
+* [Insufficient Session-ID Length](https://www.owasp.org/index.php/Insufficient_Session-ID_Length)
+* [Use of hard-coded cryptographic key](https://www.owasp.org/index.php/Use_of_hard-coded_cryptographic_key)
+* [Using a broken or risky cryptographic algorithm](https://www.owasp.org/index.php/Using_a_broken_or_risky_cryptographic_algorithm)
+* [Javax.crypto.cipher API](https://docs.oracle.com/javase/8/docs/api/javax/crypto/Cipher.html)
+* ISO 18033-1:2015 – Encryption Algorithms
+* ISO 18033-2:2015 – Asymmetric Ciphers
+* ISO 18033-3:2015 – Block Ciphers
+* [NIST Key Management Guideance](http://csrc.nist.gov/groups/ST/toolkit/key_management.html)
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md]: https://csrc.nist.gov/projects/key-management

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md [hunk 17]
```diff
@@ -119,31 +118,26 @@ CWE-329: Not Using a Random IV with CBC Mode
...
-- [NIST Key Management Guideance](http://csrc.nist.gov/groups/ST/toolkit/key_management.html)
-
-## Authors and Primary Editors
-===========================
-
-Tony Hsu - hsiang\_chih\[at\]yahoo.com
+* [NIST FIPS Standards](http://csrc.nist.gov/publications/PubsFIPS.html)
+* [Wikipedia: Initialization Vector](https://en.wikipedia.org/wiki/Initialization_vector)
+* [Secure Coding - Generating Strong Random Numbers](https://www.securecoding.cert.org/confluence/display/java/MSC02-J.+Generate+strong+random+numbers)
+* [Optimal Asymmetric Encryption Padding](http://en.wikipedia.org/wiki/Optimal_asymmetric_encryption_padding)
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md]: https://en.wikipedia.org/wiki/Optimal_asymmetric_encryption_padding

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md [hunk 18]
```diff
@@ -119,31 +118,26 @@ CWE-329: Not Using a Random IV with CBC Mode
...
-- ISO 18033-1:2015 – Encryption Algorithms
-- ISO 18033-2:2015 – Asymmetric Ciphers
-- ISO 18033-3:2015 – Block Ciphers
-- [NIST Key Management Guideance](http://csrc.nist.gov/groups/ST/toolkit/key_management.html)
-
-## Authors and Primary Editors
-===========================
-
-Tony Hsu - hsiang\_chih\[at\]yahoo.com
+* [NIST FIPS Standards](http://csrc.nist.gov/publications/PubsFIPS.html)
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.4_Testing_for_Weak_Encryption_OTG-CRYPST-004.md]: https://csrc.nist.gov/publications/fips

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md (hunk 14 above)
Comment (ThunderSon) [reply document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md]: This is different. They are not duplicates. These are references that appear on the `[5]` call

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md [hunk 19]
```diff
@@ -1180,18 +1180,18 @@ The Surf Jacking attack [[7]] was first presented by Sandro Gauci and permits to
...
+- The victim's browser starts a new clear text connection to `http://somesecuresite/` and sends an HTTP request containing the cookie in the HTTP header in clear text
+- The attacker sees this traffic and logs the cookie for later use.
 
 To test if a website is vulnerable carry out the following tests:
 
 1. Check if website supports both HTTP and HTTPS protocols
-2. Check if cookies do not have the “Secure” flag
+2. Check if cookies do not have the `Secure` flag
 
 #### SSL Strip
```
Comment (rejahrehim) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.1_Testing_for_Weak_SSL_TLS_Ciphers_Insufficient_Transport_Layer_Protection_OTG-CRYPST-001.md]: `localhost:9999` at line 1212

**Comment by ThunderSon** [comment]
Thanks @rejahrehim . As discussed, the references and inline links will be tackled at the next stage, as there are positions where there are multiple links, and we need to update and ensure that they fit the topic discussed.
All other points have been tackled.

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10_Testing_for_Weak_Cryptography.md [hunk 20]
```diff
@@ -1,4 +1,4 @@
-## 4.10 Testing for weak Cryptography
+# 4.10 Testing for weak Cryptography
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10_Testing_for_Weak_Cryptography.md]: Could you put a cap on "Weak" please?

File: document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md [hunk 21]
```diff
@@ -1,3 +1,5 @@
+# Testing for Sensitive Information Sent Via Unencrypted Channels
```
Comment (kingthorin) [inline document/4_Web_Application_Security_Testing/4.10_Testing_for_Weak_Cryptography/4.10.3_Testing_for_Sensitive_Information_Sent_via_Unencrypted_Channels_OTG-CRYPST-003.md]: Could you drop the cap on "via"?

https://capitalizemytitle.com/#Chicago

**Pull request merged**